src/onchain_reading.ts

```diff
@@ -0,0 +1,105 @@
+import {
+  decodeOrders,
+  decodeIndexedOrders,
+  Order,
+  IndexedOrder,
+} from "./encoding";
+import { BatchExchangeViewer } from "../build/types/BatchExchangeViewer";
+import { BatchExchange } from "../build/types/BatchExchange";
+import BN from "bn.js";
+
+/**
+ * Returns an iterator yielding an item for each page of order in the orderbook that is currently being collected.
+ * @param {BatchExchangeViewer} contract to query from
+ * @param {number} pageSize the number of items to fetch per page
+ * @param {number?} blockNumber the block number to execute the query at, defaults to "latest" if omitted
```

Comment:
Nit: Does this style of comment still make sense for typescript? The type of the parameter is already in the signature.

Reply:
Agree.